homeassistant/components/isy994/sensor.py

```diff
@@ -45,49 +46,51 @@ async def async_setup_entry(
 class ISYSensorEntity(ISYNodeEntity):
     """Representation of an ISY994 sensor device."""
 
-    @property
-    def raw_unit_of_measurement(self) -> str:
+    def get_raw_unit_of_measurement(
+        self, allow_special: bool = False
+    ) -> Union[dict, str]:
         """Get the raw unit of measurement for the ISY994 sensor device."""
         uom = self._node.uom
 
         # Backwards compatibility for ISYv4 Firmware:
         if isinstance(uom, list):
             return UOM_FRIENDLY_NAME.get(uom[0], uom[0])
+
+        # Special cases for ISY UOM index units:
+        isy_states = UOM_TO_STATES.get(uom)
+        if isy_states:
+            return isy_states if allow_special else None
+
         return UOM_FRIENDLY_NAME.get(uom)
 
     @property
     def state(self) -> str:
         """Get the state of the ISY994 sensor device."""
-        if self.value == ISY_VALUE_UNKNOWN:
-            return STATE_UNKNOWN
+        value = self._node.status
+        if value == ISY_VALUE_UNKNOWN:
+            return None
 
-        uom = self._node.uom
-        # Backwards compatibility for ISYv4 Firmware:
-        if isinstance(uom, list):
-            uom = uom[0]
-        if not uom:
-            return STATE_UNKNOWN
-
-        states = UOM_TO_STATES.get(uom)
-        if states and states.get(self.value):
-            return states.get(self.value)
-        if self._node.prec and int(self._node.prec) != 0:
-            str_val = str(self.value)
-            int_prec = int(self._node.prec)
-            decimal_part = str_val[-int_prec:]
-            whole_part = str_val[: len(str_val) - int_prec]
-            val = float(f"{whole_part}.{decimal_part}")
-            raw_units = self.raw_unit_of_measurement
-            if raw_units in (TEMP_CELSIUS, TEMP_FAHRENHEIT):
-                val = self.hass.config.units.temperature(val, raw_units)
-            return val
-        return self.value
+        # Get the translated ISY Unit of Measurement
+        uom = self.get_raw_unit_of_measurement(allow_special=True)
```

Comment:
It would be a bit cleaner and less error prone to have two separate functions (even if one called the other under the hood) instead of the `allow_special=True`

Reply:
Reverted

Reply:
If its a property it probably should be `raw_unit_of_measurement` instead of `get_raw_unit_of_measurement`

Reply:
Forgot to change that back